frontends/ol-components/src/components/LearningResourceExpanded/LearningResourceExpandedV2.tsx

```diff
@@ -194,11 +207,86 @@ const OnPlatform = styled.span({
   color: theme.custom.colors.black,
 })
 
-const ListButtonContainer = styled.div({
+const ButtonContainer = styled.div({
   display: "flex",
+  width: "100%",
   gap: "8px",
   flexGrow: 1,
-  justifyContent: "flex-end",
+  justifyContent: "center",
+})
+
+const StyledButton = styled(Button)<{ filled?: number }>((props) => {
```

Comment:
@gumaerc `Button` has a lot of styles built in, `StyledButton` overrides them, and then those are further overidden by `CopyLinkButton`. I'm a little worried this will become hard to keep track of.

In Figma, this looks like a new button variant `"bordered"`:

<img width="612" alt="Screenshot 2024-12-12 at 10 20 50 AM" src="https://github.com/user-attachments/assets/879e5c12-aa6a-4a44-811f-e7733888fe98" />

Figma also shows the button having a "selected" state:
<img width="549" alt="Screenshot 2024-12-12 at 10 20 58 AM" src="https://github.com/user-attachments/assets/c67c0710-01c9-46cb-bc29-d1c2b1123338" />

I don't see "selected" state details on the [main button page ](https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=13735-67429&node-type=symbol&m=dev) and I'm not sure what the selected state should look like for other variants. So, **my suggestion is:**
- Add `"bordered"` as a variant for the main button component
- Add a `selected` prop locally in `LearningResourceExpanded2.tsx`.

I took a quick stab and came up with:
```diff
diff --git a/frontends/ol-components/src/components/Button/Button.tsx b/frontends/ol-components/src/components/Button/Button.tsx
index d30d7c6ac..1b21e0efc 100644
--- a/frontends/ol-components/src/components/Button/Button.tsx
+++ b/frontends/ol-components/src/components/Button/Button.tsx
@@ -10,6 +10,7 @@ type ButtonVariant =
   | "primary"
   | "secondary"
   | "tertiary"
+  | "bordered"
   | "text"
   | "noBorder"
   | "inverted"
@@ -98,7 +99,7 @@ const buildStyles = (props: ButtonStyleProps & { theme: Theme }) => {
     ...props,
   }
   const { colors } = theme.custom
-  const hasBorder = variant === "secondary"
+  const hasBorder = variant === "secondary" || variant === "bordered"
   return css([
     {
       color: theme.palette.text.primary,
@@ -142,11 +143,6 @@ const buildStyles = (props: ButtonStyleProps & { theme: Th
me }) => {
         boxShadow: "none",
       },
     },
-    hasBorder && {
-      backgroundColor: "transparent",
-      borderColor: "currentcolor",
-      borderStyle: "solid",
-    },
     variant === "success" && {
       backgroundColor: colors.darkGreen,
       color: colors.white,
@@ -163,13 +159,10 @@ const buildStyles = (props: ButtonStyleProps & { theme: Theme }) => {
         boxShadow: "none",
       },
     },
-    hasBorder && {
+    variant === "secondary" && {
       backgroundColor: "transparent",
       borderColor: "currentcolor",
       borderStyle: "solid",
-    },
-    variant === "secondary" && {
-      color: colors.red,
       ":hover:not(:disabled)": {
         backgroundColor: tinycolor(colors.brightRed).setAlpha(0.06).toString(),
       },
@@ -217,6 +210,20 @@ const buildStyles = (props: ButtonStyleProps & { theme: Theme }) => {
       borderColor: colors.mitRed,
       borderStyle: "solid",
     },
+    variant === "bordered" && {
+      borderStyle: "solid",
+      borderColor: colors.silverGrayLight,
+      color: colors.silverGrayDark,
+      backgroundColor: "transparent",
+      ":hover:not(:disabled)": {
+        color: colors.darkGray2,
+        backgroundColor: colors.lightGray2,
+      },
+      ":disabled": {
+        border: colors.lightGray2,
+        backgroundColor: colors.lightGray2,
+      },
+    },
     // edge
     edge === "rounded" && {
       borderRadius: "4px",
```

```tsx
// In LearningResourceExpanded

const SelectableButton = styled(Button)<{
  selected?: boolean
  variant: "bordered"
}>(({ selected, theme, variant }) => [
  variant === "bordered" &&
    selected && {
      backgroundColor: theme.custom.colors.red,
      color: theme.custom.colors.white,
      borderColor: theme.custom.colors.red,
    },
])
```

Reply:
Yea, this makes sense and I kind of figured it was the case. However, @sovsey @mbilalmughal, it would be nice if you could directly notify us when you are adding new button variants so we can create a separate issue to add them to the codebase first, rather than having to shoehorn them into unrelated PR's such as this one.